synapseclient/api/agent_services.py

```diff
@@ -0,0 +1,259 @@
+
+
+async def get_session(
+    id: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Gets information about an existing chat session.
+
+    Arguments:
+        id: The ID of the session to get.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The requested session matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentSession.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(uri=f"/agent/session/{id}")
+
+
+async def update_session(
+    id: str,
+    access_level: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Updates the access level for a chat session.
+
+    Arguments:
+        id: The ID of the session to update.
+        access_level: The access level of the agent.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Request matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/UpdateAgentSessionRequest.html>
+    request = {
+        "sessionId": id,
+        "agentAccessLevel": access_level,
+    }
+    return await client.rest_put_async(
+        uri=f"/agent/session/{id}", body=json.dumps(request)
+    )
+
+
+async def send_prompt(
+    id: str,
+    prompt: str,
+    enable_trace: bool = False,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Sends a prompt to an agent starting an asyncronous job.
+
+    Arguments:
+        id: The ID of the session to send the prompt to.
+        prompt: The prompt to send to the agent.
+        enable_trace: Whether to enable trace for the prompt. Defaults to False.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The response matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/asynch/AsyncJobId.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Request matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentChatRequest.html>
+    request = {
+        "concreteType": "org.sagebionetworks.repo.model.agent.AgentChatRequest",
```

Comment:
There is a constant file to throw this in to:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/core/constants/concrete_types.py